### PR TITLE
feat(data): add API record id to Storage file prefix for API + Storage docs; update comments; update callout

### PR DESCRIPTION
**Pull request by david-mcafee**
#### Description of changes:
feat(data): add API record id to Storage file prefix for API + Storage docs; update comments; update callout

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [x] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/fragments/lib/graphqlapi/js/working-with-files.mdx [hunk 1]
```diff
@@ -47,52 +47,102 @@ For the complete working example, including required imports, obtaining the file
 
 ## Create a record with an associated file
 
-First upload the file to Storage, then run the following mutation with the GraphQL API to create a record and associate the file with the record.
+First create the API record, then upload the file to Storage, and finally add the association between the record and file. Use the following example with the GraphQL API and Storage categories to create a record and associate the file with the record.
```
Comment (renebrandel) [inline src/fragments/lib/graphqlapi/js/working-with-files.mdx:50]:
```suggestion
First create a record via the GraphQL API, then upload the file to Storage, and finally add the association between the record and file. Use the following example with the GraphQL API and Storage categories to create a record and associate the file with the record.
```

File: src/fragments/lib/graphqlapi/js/working-with-files.mdx [hunk 2]
```diff
@@ -47,52 +47,102 @@ For the complete working example, including required imports, obtaining the file
...
 
 <Callout>
 
-Note: In the case of adding a file to an existing record (such as when a user uploads a profile picture to an existing profile), first create the record and exclude the optional file reference, then follow the steps under [Add or update a file for an associated record](#add-or-update-a-file-for-an-associated-record) below.
+The API record's `id` is prepended to the Storage file name to ensure uniqueness. If this is excluded, multiple API records could then be associated with the same file key unintentionally.
```
Comment (renebrandel) [inline src/fragments/lib/graphqlapi/js/working-with-files.mdx:54]:
```suggestion
Prepend a record's `id` to the Storage file name to ensure uniqueness. If this is excluded, multiple records could then be associated with the same file key unintentionally.
```

**Review by renebrandel — APPROVED** [review]

File: src/fragments/lib/graphqlapi/js/working-with-files.mdx [hunk 3]
```diff
@@ -584,6 +680,7 @@ function App() {
         id: _song.id,
       };
 
+      // const deletedSong = await API.graphql<GraphQLQuery<DeleteSongMutation>>({
```
Comment (iartemiev) [inline src/fragments/lib/graphqlapi/js/working-with-files.mdx:683]: Is this line needed?

File: src/fragments/lib/graphqlapi/js/working-with-files.mdx [hunk 4]
```diff
@@ -47,52 +47,102 @@ For the complete working example, including required imports, obtaining the file
...
+  name: `My first song`,
+};
+
+// Create the API record:
+const response = await API.graphql<GraphQLQuery<CreateSongMutation>>({
+  query: mutations.createSong,
+  variables: { input: createSongDetails },
 });
 
+const _song = response?.data?.createSong;
```
Comment (iartemiev) [inline src/fragments/lib/graphqlapi/js/working-with-files.mdx:72]: Why is this variable prepended with an underscore? Is there something special about it vs all the others?

File: src/fragments/lib/graphqlapi/js/working-with-files.mdx [hunk 5]
```diff
@@ -47,52 +47,102 @@ For the complete working example, including required imports, obtaining the file
...
+const updateResponse = await API.graphql<
+  GraphQLQuery<UpdateSongMutation>
+>({
+  query: mutations.updateSong,
+  variables: { input: updateSongDetails },
 });
+
+const updatedSong = updateResponse?.data?.updateSong;
+
+// Check that the record has an associated image:
```
Comment (iartemiev) [inline src/fragments/lib/graphqlapi/js/working-with-files.mdx:96]: Nit
```suggestion
// Ensure the record has an associated image:
```

**Review by iartemiev — DISMISSED** [review]
Looks good, just some minor call outs

File: src/fragments/lib/graphqlapi/js/working-with-files.mdx (hunk 3 above)
Comment (david-mcafee) [reply src/fragments/lib/graphqlapi/js/working-with-files.mdx:683]: Nope, will remove.

File: src/fragments/lib/graphqlapi/js/working-with-files.mdx (hunk 4 above)
Comment (david-mcafee) [reply src/fragments/lib/graphqlapi/js/working-with-files.mdx:72]: I'd previously been using this naming convention, and later switched. Will make this consistent with the other variables.

Comment (david-mcafee) [reply src/fragments/lib/graphqlapi/js/working-with-files.mdx:72]: This is not needed, will update.

**Review by renebrandel — APPROVED** [review]

**Review by katiegoines — APPROVED** [review]

**Pull request merged**